### PR TITLE
add qa tutorial

### PR DESCRIPTION
**Pull request by yzhang123**
Signed-off-by: Yang Zhang <yangzhang@nvidia.com>

### CODE REVIEW

File: docs/source/nlp/question_answering.rst [hunk 1]
```diff
@@ -0,0 +1,339 @@
+.. _token_classification:
```
Comment (ekmb) [inline docs/source/nlp/question_answering.rst:1]: should be a different name to be tagged in the docs

File: docs/source/nlp/question_answering.rst [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+| optim.sched.name                          | string          | SquareRootAnnealing                                                              | The warmup schedule                                                                                          |
++-------------------------------------------+-----------------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------+
+| optim.sched.warmup_ratio                  | float           | 0.0                                                                              | The warmup ratio                                                                                             |
++-------------------------------------------+-----------------+----------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------+
```
Comment (ekmb) [inline docs/source/nlp/question_answering.rst:218]: how about removing max_epochs, batch_size, shuffle and optim related params since they are not model specific?

File: docs/source/nlp/question_answering.rst [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+^^^^^^^^^^^^^^^^^^^
+* :code:`pretrained_model`: pretrained QAModel model from list_available_models() or path to a .nemo file
+* :code:`model.test_ds.file`: Path to test file.
+
+During evaluation of the :code:`test_ds`, the script generates the following metrics:
+
+* :code:`Exact Match (EM)`
+* :code:`F1`
+
+More details about these metrics could be found `here <https://en.wikipedia.org/wiki/Precision_and_recall>`__.
```
Comment (ekmb) [inline docs/source/nlp/question_answering.rst:331]: wrong link

File: docs/source/nlp/question_answering.rst [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+
+    While the arguments are defined in the spec file, if you wish to override these parameter definitions in the spec file \
+    and experiment with them, you may do so over command line by simple defining the param. \
+    For example, the sample spec file mentioned above has :code:`validation_ds.batch_size` set to 24. \
+    However, if you see that the GPU utilization can be optimized further by using larger a batch size, \
+    you may override to the desired value, by adding the field :code:`validation_ds.batch_size=128` over command line.
+    You may repeat this with any of the parameters defined in the sample spec file.
+
+Important parameters
+^^^^^^^^^^^^^^^^^^^^
```
Comment (ekmb) [inline docs/source/nlp/question_answering.rst:257]: I have the same passage for other models, maybe move this to a separate NLPModel .rst along with other common params? Could you please add it?

**Review by ekmb — CHANGES_REQUESTED** [review]

File: docs/source/nlp/question_answering.rst (hunk 2 above)
Comment (yzhang123) [reply docs/source/nlp/question_answering.rst:218]: ok

File: docs/source/nlp/question_answering.rst (hunk 4 above)
Comment (yzhang123) [reply docs/source/nlp/question_answering.rst]: ok

**Comment by yzhang123** [comment]
Cannot root cause why DCO fails

File: docs/source/nlp/question_answering.rst [hunk 5]
```diff
@@ -0,0 +1,268 @@
+.. _question_answering:
+
+Question Answering Model
+=====================================================
+
```
Comment (ekmb) [inline docs/source/nlp/question_answering.rst:5]: is it compiling fine? it usually throws a warning then `==` is longer than the title

File: docs/source/nlp/question_answering.rst [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+- The likelihood this word is the end of the span
+
+The model chooses the start and end words with maximal probabilities. When the content does not
+contain the answer, we would like the start and end span to be set for the first token.
+
+A pretrained BERT encoder with two span prediction heads is used for the prediction start and
+the end position of the answer. The span predictions are token classifiers consisting of a single
+linear layer.
+
+
```
Comment (ekmb) [inline docs/source/nlp/question_answering.rst:24]: If there are pre-trained models available, could you add a quick start like here https://github.com/NVIDIA/NeMo/blob/r1.0.0rc1/docs/source/nlp/punctuation_and_capitalization.rst#quick-start

File: docs/source/nlp/question_answering.rst [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+    We recommend you try this model in a Jupyter notebook \
+    (can run on `Google's Colab <https://colab.research.google.com/notebooks/intro.ipynb>`_.): \
+    `NeMo/tutorials/nlp/Question_Answering_Squad.ipynb <https://github.com/NVIDIA/NeMo/blob/main/tutorials/nlp/Question_Answering_Squad.ipynb>`__.
+
+    Connect to an instance with a GPU (Runtime -> Change runtime type -> select "GPU" for hardware accelerator)
+
+    An example script on how to train and evaluate the model could be found here: `NeMo/examples/nlp/question_answering/question_answering_squad.py <https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/question_answering/question_answering_squad.py>`__.
+
+    The default configuration file for the model could be found at: `NeMo/examples/nlp/token_classification/conf/token_classification_config.yaml <https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/question_answering/conf/question_answering_squad_config.yaml>`__.
```
Comment (ekmb) [inline docs/source/nlp/question_answering.rst:35]: incorrect link

File: docs/source/nlp/question_answering.rst [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+
+
+
+
+.. _dataset_question_answering:
+
+Data Format
+-----------------------------------------
```
Comment (ekmb) [inline docs/source/nlp/question_answering.rst:43]: format

File: docs/source/nlp/question_answering.rst [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+         |-- v1.1/dev-v1.1.json
+         |-- v2.0/train-v2.0.json
+         |-- v2.0/dev-v2.0.json
+
+
+.. _model_training_question_answering:
+
+Model Training
+-----------------------------------
+
```
Comment (ekmb) [inline docs/source/nlp/question_answering.rst:135]: format

**Review by ekmb — CHANGES_REQUESTED** [review]

File: docs/source/nlp/question_answering.rst (hunk 5 above)
Comment (yzhang123) [reply docs/source/nlp/question_answering.rst:5]: works for me

File: docs/source/nlp/question_answering.rst (hunk 6 above)
Comment (yzhang123) [reply docs/source/nlp/question_answering.rst:24]: they are not available yet

File: docs/source/nlp/question_answering.rst (hunk 8 above)
Comment (yzhang123) [reply docs/source/nlp/question_answering.rst:43]: whats the issue here?

File: docs/source/nlp/question_answering.rst (hunk 9 above)
Comment (yzhang123) [reply docs/source/nlp/question_answering.rst:135]: same as above

**Review by ekmb — APPROVED** [review]
